modules/gnetBidAdapter.js

```diff
@@ -0,0 +1,111 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import * as utils from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'gnet';
+const ENDPOINT = 'https://grumft.cbykmedia.com/prebid/prebid.php';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    return !!(bid.params.websiteId && bid.params.externalId);
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const bidRequests = [];
+    const referer = bidderRequest.refererInfo.referer;
+
+    utils._each(validBidRequests, (request) => {
+      const data = Object();
+
+      data.referer = referer;
+      data.adUnitCode = request.adUnitCode;
+      data.bidId = request.bidId;
+      data.transactionId = request.transactionId;
+
+      data.sizes = [];
+      utils._each(request.sizes, (size) => {
+        data.sizes.push(size[0] + 'x' + size[1]);
+      });
```

Comment:
there is a util function which I believe does this for you if you want to use it.

`utils. parseSizesInput(request.sizes)`

It is recommended to look into the mediaType.banner.sizes instead of the top level `sizes` attribute though.

We did "deprecate" top level sizes on adUnits, but maintained it for use for bid adapters. So, it is just a recommendation if you do not wish to use it.

Reply:
Nice! I'll change it.. Thanks!